### PR TITLE
build(release): add pre-release workflow to create version bump PR

### PR DESCRIPTION
**Pull request by hairyhum**
## Change Overview

Currently release process is happening outside of github workflows. This is a first step to move it there.

Pre-release workflow can be run from workflow dispatch in github actions and will create a PR with version update.

I've also changed the version bump PR title because it's not only updating docs

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [x] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

Test runs can be found in https://github.com/kanisterio/kanister/actions/workflows/pre-release.yml

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by hairyhum** [comment]
Added RELEASE.md with kanister release instructions. It currently describes the process with reno release notes, which are not merged yet.

**Comment by hairyhum** [comment]
Rebased it on `reno-changelog` branch

File: .github/workflows/pre-release.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - name: create_pr_body
+      run: |
+        echo "Update version to ${RELEASE_TAG}" > PR_BODY_FILE
+        echo "" >> PR_BODY_FILE
+        echo "Please check the changelog for the following merges:" >> PR_BODY_FILE
+        export CURRENT_TAG=$(git describe --abbrev=0 --tags)
+        git log ${CURRENT_TAG}..kan-docs-${PRERELEASE_DOCS_BRANCH}-${RELEASE_TAG} --pretty="- %h: %s"  | grep -v ': test' | grep -v ': doc' | grep -v ': build' | grep -v ': deps'  >> PR_BODY_FILE
+    - name: create_pr
+      run: |
+        gh pr create --title "pre-release: Update version to ${RELEASE_TAG}" -F PR_BODY_FILE --head "kan-docs-${PRERELEASE_DOCS_BRANCH}-${RELEASE_TAG}" --base master --reviewer tdmanv,pavannd1,viveksinghggits,hairyhum --label kueue
```
Comment (pavannd1) [inline .github/workflows/pre-release.yml:54]:
```suggestion
        gh pr create --title "pre-release: Update version to ${RELEASE_TAG}" -F PR_BODY_FILE --head "kan-docs-${PRERELEASE_DOCS_BRANCH}-${RELEASE_TAG}" --base master --reviewer pavannd1,viveksinghggits,hairyhum --label kueue
```
[nit] Let's start with the three of us for now

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
+## Automated Release
```
Comment (pavannd1) [inline RELEASE.md:1]: @hairyhum please remove all the Kasten K10 related information from this file. Maybe don't add this file until we move the entire release process to GH.

Comment (hairyhum) [reply RELEASE.md]: Moved to https://github.com/kanisterio/kanister/pull/2829

File: .github/workflows/pre-release.yml
Comment (viveksinghggits) [inline .github/workflows/pre-release.yml:1]: should we have a document somewhere, that can help someone trigger the pre-release/release process?

**Review by viveksinghggits — COMMENTED** [review]
LGTM

File: .github/workflows/pre-release.yml
Comment (hairyhum) [reply .github/workflows/pre-release.yml:1]: The doc is moved to release workflow PR https://github.com/kanisterio/kanister/pull/2829 to get all the pieces.

File: .github/workflows/pre-release.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        description: 'Image tag in the format x.x.x'
+        required: true
+        type: string
+
+env:
+  RELEASE_TAG: ${{ inputs.release_tag }}
+  PRERELEASE_DOCS_BRANCH: 'dg8d45z'
+
+jobs:
+  ## TODO we can add a condition like github.actor.role == 'Maintainer' to limit trigger to maintainers only
```
Comment (viveksinghggits) [inline .github/workflows/pre-release.yml:16]: should we do this as part of this PR itself? I think we should.

Comment (hairyhum) [reply .github/workflows/pre-release.yml:16]: This is not yet supported, but there's active discussion ongoing. When it's implemented, we can do that.

**Review by viveksinghggits — APPROVED** [review]

**Review by e-sumin — APPROVED** [review]
LGTM

**Pull request merged**